source/common/http/http1/codec_impl.h

```diff
@@ -344,19 +347,26 @@ class ConnectionImpl : public virtual Connection,
   void onDispatch(const Buffer::Instance& data);
 
   // ParserCallbacks.
-  Status onHeaderField(const char* data, size_t length) override;
-  Status onHeaderValue(const char* data, size_t length) override;
-  Envoy::StatusOr<ParserStatus> onHeadersComplete() override;
+  ParserStatus onMessageBegin() override;
+  ParserStatus onUrl(const char* data, size_t length) override;
+  ParserStatus onStatus(const char* data, size_t length) override;
+  ParserStatus onHeaderField(const char* data, size_t length) override;
+  ParserStatus onHeaderValue(const char* data, size_t length) override;
+  ParserStatus onHeadersComplete() override;
   void bufferBody(const char* data, size_t length) override;
-  StatusOr<ParserStatus> onMessageComplete() override;
-  ParserStatus setAndCheckCallbackStatus(Http::Status&& status) override;
-  ParserStatus setAndCheckCallbackStatusOr(Envoy::StatusOr<ParserStatus>&& statusor) override;
-
-  // Connection specific callback methods.
-  virtual Envoy::StatusOr<ParserStatus> onHeadersCompleteBase() PURE;
-  virtual ParserStatus onMessageCompleteBase() PURE;
```

Comment:
Re: moving the method from private -> protected, it weakens encapsulation as there's now a larger swath of potential callers. I think this is a common enough pattern in C++. 

From Google C++ styleguide: Limit the use of protected to those member functions that might need to be accessed from subclasses. in https://google.github.io/styleguide/cppguide.html#Inheritance

Perhaps onMessageBeginningBase should become consistent with these